go/ql/lib/semmle/go/Types.qll

```diff
@@ -747,6 +768,33 @@ class InterfaceType extends @interfacetype, CompositeType {
     exists(int i | i >= 0 | component_types(this, i, name, result))
   }
 
+  /**
+   * Gets the type of method `qname` of this interface type.
+   *
+   * This differs from `getMethodType` in that if the method is not exported, the `qname`
+   * will be package-qualified. This means that the set of `qname`s` together with any
+   * embedded types fully distinguishes the interface from any other, whereas the set
+   * of names matched by `getMethodName` may be ambiguous between interfaces with matching
+   * exported methods and unexported methods that have matching names but belong to
+   * different packages.
```

Comment:
It's confusing that you sometimes get the qualified name and sometimes the unqualified name. How about giving the qualified name in both cases? That would match the predicate name.

Reply:
Not done: the goal here is to only apply qualifiers where needed to distinguish types, i.e. if the method is non-exported and declared in a different package.

Reply:
Then maybe the predicate name should be changed from `getMethodTypeByQualifiedName`? At the moment it's just a recipe for confusion. Distinguishing name? Possibly qualified name?

Reply:
I think I agree with @owen-mc here. It may be that the motivation for adding this was to distinguish types that otherwise appear equal, but that doesn't mean it has to be limited to the minimum necessary to accomplish that at the cost of having a confusingly-named predicate.

If the intention is to implement something just distinguish between different interfaces, I'd maybe call this something like `getIdentityMethodName`?

Also, the comment here could be clearer. There is some ambiguity in what "both" in "would yield ... for both" refers to. It's also not immediately obvious from the documentation how it relates to how the predicate is used.

Reply:
I've simply deleted the predicate and moved the complexity to the user: https://github.com/github/codeql/pull/17357/commits/87c1e7decf83caffcee83771f12b1587f30b31f1